src/NServiceBus.Core.Tests/Routing/NamespaceRouteSourceTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System.Linq;
+    using System.Reflection;
+    using NamespaceRouteSourceTest;
+    using NamespaceRouteSourceTest.OtherNamespace;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class NamespaceRouteSourceTests
+    {
+        [Test]
+        public void It_returns_only_message_types()
+        {
+            var source = new NamespaceRouteSource(Assembly.GetExecutingAssembly(), "NServiceBus.Core.Tests.Routing.NamespaceRouteSourceTest", UnicastRoute.CreateFromEndpointName("Destination"));
+            var routes = source.GenerateRoutes(new Conventions()).ToArray();
+
+            Assert.IsTrue(routes.Any(r => r.MessageType == typeof(Message)));
```

Comment:
I general I prefer Collection asserts since they provide usually better assert messages when the test fails than Assert.IsTrue / IsFalse on collections. @andreasohlund seems to agree